### PR TITLE
Normalize CPU vol 2

### PR DESCRIPTION
**Pull request by mzient**
#### Why we need this PR?
*Pick one, remove the rest*
- It adds a CPU kernel for normalizing tensors (`(x-mean)*(1/stddev)`) with arbitrary axis broadcasting

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Use zero strides to unify broadcast vs non-broadcast axes
     * Squeeze dimensions to reduce nesting level
 - Affected modules and functionalities:
     * kernels/normalize
 - Key points relevant for the review:
     * normalize_cpu.h
 - Validation and testing:
     * Unit tests, up to 6D, with exhaustive axis subsets
 - Documentation (including examples):
     * Doxygen

**JIRA TASK**: DALI-1242

### CODE REVIEW

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

**Comment by mzient** [comment]
!build

File: dali/kernels/normalize/normalize_cpu.h [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        normalize(out, in, data_shape_[axis], mean, stddev);    // per-element factors
+    } else if (axis == ndim() - 2) {
+      // 2D case can be either normalizing the inner or the outer dimension
+      if (param_strides[axis] == 0 && param_strides[axis+1] != 0) {
+        // e.g. normalize R,G,B interleaved channels using per-channel normalization factors
+        // shared across pixels
+        assert(param_strides[axis+1] == 1);
+        normalize_inner(out, in, data_shape_[axis], data_shape_[axis+1], mean, stddev);
+      } else {
+        assert(param_strides[axis] ==1 && param_strides[axis+1] == 0);
```
Comment (banasraf) [inline dali/kernels/normalize/normalize_cpu.h]:
```suggestion
        assert(param_strides[axis] == 1 && param_strides[axis+1] == 0);
```
;)

Comment (mzient) [reply dali/kernels/normalize/normalize_cpu.h]: Will do in operator PR?

**Pull request merged**